File: accounts/accounts-domain/src/main/java/com/hrrm/famoney/accounts/movement/EntryItem.java
package com.hrrm.famoney.accounts.movement;

import lombok.*;
import lombok.experimental.Accessors;
import org.jetbrains.annotations.NotNull;
import org.jetbrains.annotations.Nullable;

import javax.persistence.Column;
import javax.persistence.Embeddable;
import java.math.BigDecimal;

@Embeddable
@Accessors(chain = true)
@Getter
@Setter
@ToString(callSuper = true)
@EqualsAndHashCode(onlyExplicitlyIncluded = true)
public class EntryItem {

    @Column(name = "pos")
    @NotNull
    private Integer position;

    @Column(name = "category_id")
    @NotNull
    private Integer categoryId;

    @Column(name = "amount")
    @NotNull
    private BigDecimal amount;

    @Column(name = "comments")
    @Nullable
    private String comments;

}

File: accounts/accounts-domain/src/main/kotlin/com/hrrm/famoney/accounts/movement/EntryItem.kt
package com.hrrm.famoney.accounts.movement

import lombok.experimental.Accessors
import java.math.BigDecimal
import javax.persistence.Column
import javax.persistence.Embeddable

@Embeddable
@Accessors(chain = true)
class EntryItem {
    @Column(name = "pos")
    var position: Int = 0

    @Column(name = "category_id")
    var categoryId: Int = 0

    @Column(name = "amount")
    var amount: BigDecimal = BigDecimal.ZERO

    @Column(name = "comments")
    var comments: String? = null
}